JIP-4.md

```diff
@@ -0,0 +1,28 @@
+# JIP-4: Chainspec file
+
+A chain specification collects information that describes a JAM-based network. It identifies the network a blockchain node connects to, the other nodes it initially communicates with, and the initial state that nodes must agree on to produce blocks. The chain specification can be defined in a JSON file.
+
+## Chain Specification JSON Format
+
+The chain specification root object includes the following keys:
+
+- `id` - the machine-readable identifier for the network. This may be used as part of the network protocol identifier in the future version of the network protocol.
+- `bootnodes` - an optional list of the nodes accepting connections. Each entry is a string in the following format: `<name>@<ip>:<port>` where `<name>` is the 53-character DNS name consisting of "e" followed by the Ed25519 public key, base-32 encoded using the alphabet "abcdefghijklmnopqrstuvwxyz234567". `<ip>` is a string containing IPv4 or IPv6 address of the node. IPv6 address may optionally be specified in square brackets (`[]`). `<port>` is an IP port number.
+- `genesis_header` - A hex string containing JAM-serialized genesis block header.
```

Comment:
what if making `genesis_header` human readable while it is for the `spec`, otherwise, patch the data structure in this JIP in the following sections that people can read it more efficiently

Reply:
See https://github.com/polkadot-fellows/JIPs/pull/1#issuecomment-2883087286